Ark.Tools.Outbox/IOutboxAsyncProducer.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Ark.Tools.Outbox
+{
+    /// <summary>
+    /// Interface for a message Producer used by Application to submit messages to the Outbox
+    /// </summary>
+    public interface IOutboxAsyncService
+    {
+        Task PublishAsync<T>(IOutboxContextAsync ctx, T message, CancellationToken ctk = default) where T : class
+#if NETSTANDARD2_1
+            => PublishAsync(ctx, message, null, ctk)
+#endif
+            ;
+
+        Task PublishAsync<T>(IOutboxContextAsync ctx, T message, IDictionary<string, string>? optionalHeaders = null, CancellationToken ctk = default) where T : class;
+    }
+
+    public interface IOutboxAsyncConsumer
+    {
+        Task StartAsync(CancellationToken ctk = default);
+        Task StopAsync(CancellationToken ctk = default);
+
+        Task ClearAsync(CancellationToken ctk = default);
+        Task<int> CountAsync(CancellationToken ctk = default);
+    }
+
+    public abstract class OutboxAsyncConsumerBase : IOutboxAsyncConsumer, IAsyncDisposable
+    {
+        private readonly Func<IOutboxContextAsync> _outboxContextFactory;
+        private CancellationTokenSource? _processorCT;
+        private Task? _processorTask;
+        private bool _disposedValue;
+
+        public TimeSpan SleepInterval { get; set; } = TimeSpan.FromSeconds(1);
+        public int BatchSize { get; set; } = 1000;
+
+        public OutboxAsyncConsumerBase(Func<IOutboxContextAsync> contextFactory)
+        {
+            _outboxContextFactory = contextFactory;
+        }
+
+        public async Task ClearAsync(CancellationToken ctk = default)
+        {
+            await using var ctx = _outboxContextFactory();
```

Comment:
this cannot work, Factory is now async.

this should be `await using var ctx = await _factorySomething.CreateAsync(ctk)`